File: src/main/java/net/codinux/kotlin/Strings.java
package net.codinux.kotlin;

import java.util.Locale;

public class Strings {

    public void isNotEmpty() {
        String name = "Mahatma";

        if (name.isEmpty() == false) {

        }
    }

    public void isNullOrBlank() {
        String name = null;

        if (name == null || name.isBlank()) {

        }
    }

    public void ignoreCase() {
        String name = "Mahatma";

        System.out.println(name.toLowerCase(Locale.ROOT).contains("Atma".toLowerCase(Locale.ROOT)));
    }

    public void encodeToByteArray() {
        String name = "Mahatma";

        // TODO:
        // System.out.println(name.encodeToByteArray());
    }

}

File: src/main/kotlin/net/codinux/kotlin/Strings.kt
package net.codinux.kotlin


class Strings {

  fun isNotEmpty() {
    val name = "Mahatma"

    if (name.isNotEmpty()) {

    }
  }

  fun isNullOrBlank() {
    val name: String? = null

    if (name.isNullOrBlank()) { // can even be called on null

    }
  }

  fun ignoreCase() {
    val name = "Mahatma"

    println(name.contains("Atma", ignoreCase = true)) // prints true
    // also works for .startsWith(), endsWidth(), .replace(), ...
    // "ignoreCase = " can be left away, just to make code clearer
  }

  fun encodeToByteArray() {
    val name = "Mahatma"

    println(name.encodeToByteArray())
  }

}